demos/yielding/README.md

```diff
@@ -0,0 +1,112 @@
+# Yielding
+
+In this demo, we try to show the usage of yielding, which is one characteristic of a multitasking system. 
+
+By calling syscall `yield()`, the thread gives away the utilization of a CPU to another thread, or a specific thread (will be discussed later in another demo).
+
+To achieve thread yielding, we need to introduce the following functions:
+
+* `add_thread_from_worker` (in `src/user/thread.c`)
+* `do_scheduler` (in `src/kernel/thread.c`)
+* `yield` (in `src/user/thread.c`)
+
+## `add_thread_from_worker`
+
+In AMT, we try to maintain all of the threads in an array called `all_threads`.
+
+By calling this function, the AMT will create a new thread instance with the given arguments, initialize the permissions, and put the function in. 
```

Comment:
The start is a bit odd, what about:
"When a thread calls this function, the kernel will..."

When referring to AMT, either this should be "AMT will" or "the AMT kernel will" or "the kernel will". In this case I would go with "the kernel will", just to save some words. (you're going to be saying this a lot and no one will confuse it with "the kernel" aka the Linux kernel)

"a new thread instance"

This can just be "a new thread". (it's not wrong, it is an instance but it's just not needed)

"and put the function in"

I know what you're trying to say here but it isn't clear if you don't know the system. "and set it's starting point to be the given function".
(perhaps showing the function signature would help here?)

Side note: saying "starting point" isn't quite true since it goes thread_start->user function but I think it fits fine here. From the user space point of view that's the starting point.